### PR TITLE
Add egress loss stat

### PR DESCRIPTION
**Pull request by k0nserv**
Add an egress loss stat to `PeerStats` derived from TWCC reports.

### CODE REVIEW

File: src/rtp/rtcp/twcc.rs [hunk 1]
```diff
@@ -1088,6 +1088,7 @@ impl TwccSendRegister {
             );
         }
 
+        println!("TWCC loss: {:?}", self.loss(Duration::from_secs(1), now));
```
Comment (algesten) [inline src/rtp/rtcp/twcc.rs:1091]: ?

File: src/rtp/rtcp/twcc.rs [hunk 2]
```diff
@@ -1097,6 +1098,53 @@ impl TwccSendRegister {
...
+            (total + 1, lost + u64::from(was_lost))
+        });
+
+        if total == 0 {
+            return None;
+        }
+
+        Some((lost as f64) / (total as f64))
+    }
+
```
Comment (algesten) [inline src/rtp/rtcp/twcc.rs:1147]: This looks a bit expensive. Is it?

File: src/rtp/rtcp/twcc.rs (hunk 1 above)
Comment (k0nserv) [reply src/rtp/rtcp/twcc.rs:1091]: Soz, left from debugging

File: src/rtp/rtcp/twcc.rs (hunk 2 above)
Comment (k0nserv) [reply src/rtp/rtcp/twcc.rs:1147]: Shouldn't be, it's `O(2n)` at worst with `n == 1000` currently and runs every second. I don't really think the `2n` case will happen either

File: src/stats.rs [hunk 3]
```diff
@@ -68,6 +70,8 @@ pub struct PeerStats {
     pub timestamp: Instant,
     /// The last egress bandwidth estimate from the BWE subsystem, if enabled.
     pub bwe_tx: Option<Bitrate>,
+    /// The egress loss over the last second expressed as a percentage value between 0.0 and 1.0.
```
Comment (xnorpx) [inline src/stats.rs:73]: Fraction or ratio?

Comment (xnorpx) [reply src/stats.rs:73]: or just multiply by 100 and make it an u8?

Comment (k0nserv) [reply src/stats.rs:73]: Hmm, I think keeping it as f64 would be nice but I can reword it

Comment (algesten) [reply src/stats.rs:73]: f32?

Comment (k0nserv) [reply src/stats.rs:73]: No preference, I just defaulted to f64. Do we have a stated preference for the project?

Comment (algesten) [reply src/stats.rs:73]: i think it depends on what we're talking about. a percentage number between 0.0-1.0 might not need the precision of f64.

Comment (k0nserv) [reply src/stats.rs:73]: Fair, f32 should be sufficient. Will change

Comment (xnorpx) [reply src/stats.rs:73]: also if you say egress_loss_ratio it is clear that you return a ratio and I will expect numbers between 0 and 1

Comment (k0nserv) [reply src/stats.rs:73]: > also if you say egress_loss_ratio it is clear that you return a ratio and I will expect numbers between 0 and 1

Good point, although I know @algesten isn't a fan of long names :D

File: src/rtp/rtcp/twcc.rs [hunk 4]
```diff
@@ -1811,4 +1858,56 @@ mod test {
...
+                .into(),
+            },
+            now,
+        )
+        .expect("apply_report to return Some(_)");
+
+        now = now + Duration::from_millis(20);
+        let loss = reg
+            .loss(Duration::from_millis(150), now)
+            .expect("Should be able to calcualte loss");
```
Comment (xnorpx) [inline src/rtp/rtcp/twcc.rs:1907]: calculate

File: src/stats.rs (hunk 3 above)
Comment (xnorpx) [reply src/stats.rs:73]: comment still percentage

**Comment by xnorpx** [comment]
lgtm

**Review by algesten — APPROVED** [review]
